src/Utils/StrictPlus.php

```diff
@@ -0,0 +1,233 @@
+<?php
+
+/**
+ * This file is part of the Nette Framework (https://nette.org)
+ * Copyright (c) 2004 David Grudl (https://davidgrudl.com)
+ */
+
+namespace Nette;
+
+use Nette\Utils\Callback;
+use Nette\Utils\ObjectMixin;
+
+
+/**
+ * Strict class for better experience.
+ * - 'did you mean' hints
+ * - access to undeclared members throws exceptions
+ * - support for @property annotations
+ * - support for calling event handlers stored in $onEvent via onEvent()
+ * - compatible with Nette\Object
+ */
+trait StrictPlus
```

Comment:
> **Enchantments** sounds like something from a fantasy game and isn't really describing what it does. 

I think it sort-of does. It casts a couple of spells on the given class to alter it's behavior. It's similar to BlackMagic but sounds more optimistic.